x/incentives/simulation/operations.go

```diff
@@ -62,26 +63,45 @@ func WeightedOperations(
 	}
 }
 
+// Generates a random number of coin denoms with a respective random value for each coin.
 func genRewardCoins(r *rand.Rand, coins sdk.Coins) (res sdk.Coins) {
 	numCoins := 1 + r.Intn(Min(coins.Len(), 1))
 	denomIndices := r.Perm(numCoins)
 	for i := 0; i < numCoins; i++ {
 		denom := coins[denomIndices[i]].Denom
-		amt, _ := simtypes.RandPositiveInt(r, coins[i].Amount)
+		amt, err := simtypes.RandPositiveInt(r, coins[i].Amount)
+		if err != nil {
+			panic(err)
+		}
 		res = append(res, sdk.Coin{Denom: denom, Amount: amt})
 	}
-
 	return
 }
 
+// Returns a single lockup QueryCondition, which is generated from a single coin randomly selected from the provided coin array
 func genQueryCondition(r *rand.Rand, blocktime time.Time, coins sdk.Coins, durations []time.Duration) lockuptypes.QueryCondition {
 	// TODO: reset to 2 after postlaunch, only allow duration based query type on postlaunch
 	// lockQueryType := r.Intn(2)
+	// TODO: why reset to 2 after postlaunch? Only 0 and 1 exist, and 1 is no longer used.
```

Comment:
TODO: why reset to 2 after postlaunch? Only 0 and 1 exist, and 1 is no longer used.

Reply:
r.Intn generates a random number less than n, so 0 or 1 in this case. So whats happening here is were generating just 0. And at launch, when we disabled by time, we thought it was temporary, but it turned out this was the permanent correct decision.